client/src/components/employer-dashboard-component/index.jsx

```diff
@@ -1,62 +1,158 @@
-import { useState } from "react";
 // import { useNavigate } from 'react-router-dom';
 import PropTypes from "prop-types";
 import { Box } from "@mui/material";
+import * as React from "react";
 import { Container, DashboardContainer } from "./index.styles";
 import EmployerDashboardHeader from "./employer-dashboard-header";
 import EmployerDashboardFilter from "./employer-dashboard-filter";
 import EmployerDashboardTable from "./employer-dashboard-table";
+import EmployerType from "../../prop-types/EmployerType";
+import { getFilteredEmployers } from "../../utils/api";
+import ErrorComponent from "../shared/error-screen-component";
+import LoadingScreenComponent from "../shared/loading-screen-component";
 
-function EmployerDashboardComponent({ employerData }) {
-  const [page, setPage] = useState(0);
-  const [rowsPerPage, setRowsPerPage] = useState(10);
+function EmployerDashboardComponent({ employers, setEmployers, getUserById }) {
+  const [initialLoading, setInitialLoading] = React.useState(true);
+  const [loading, setLoading] = React.useState(false);
+  const [paginationModel, setPaginationModel] = React.useState({
+    pageSize: 10,
+    page: 0,
+  });
+  const [rowCount, setRowCount] = React.useState(employers.length);
+  const [errorOb, setError] = React.useState(null);
 
-  const handleChangePage = (event, newPage) => {
-    setPage(newPage);
+  // helper to generate query params based on pagination model state and filter configs
+  const declareFilterEmployerQueryParams = (
+    filterParams,
+    customPageModel = null,
+  ) => {
+    let { pageSize, page } = paginationModel;
+    if (customPageModel) {
+      page = customPageModel.page;
+      pageSize = customPageModel.pageSize;
+      setPaginationModel(customPageModel);
+    }
+    // we initially include pagination model first
+    const queryParams = new URLSearchParams({
+      pageSize,
+      page,
+    });
+
+    // early return if no filter params are provided
+    if (!filterParams) return queryParams;
+    // ensure these filter configs are defined before passing in query
+    if (filterParams.employerName)
+      queryParams.append("employerName", filterParams.employerName);
+    if (filterParams.phoneNumber)
+      queryParams.append("phoneNumber", filterParams.phoneNumber);
+    if (filterParams.dateFrom)
+      queryParams.append("startDateAdded", filterParams.dateFrom.toISOString());
+    if (filterParams.dateTo)
+      queryParams.append("endDateAdded", filterParams.dateTo.toISOString());
```

Comment:
Date filtering isn't quite working properly atm. I think it may be because setting the iso string on the backend to time 0 0 0 0 or time 29 59 59 59 after already being converted to an iso string doesn't necessarily correspond to time 0 0 0 0 in EST and time 29 59 59 59 in EST as we were expecting it to.